### PR TITLE
shinesolutions/puppet-aem-curator#220

### PR DESCRIPTION
**Pull request by hoomaan-kh**
Added ssl configuration to packer-aem
Added JDK11 support to packer-aem
Added aem-ssl-method
Removed oracle_java module

### CODE REVIEW

File: Puppetfile [hunk 1]
```diff
@@ -4,27 +4,25 @@ mod 'bashtoni/timezone', '1.0.0'
 mod 'puppet/selinux', '0.7.1'
 mod 'kemra102/cloudwatchlogs', :git => 'https://github.com/kemra102/puppet-cloudwatchlogs', :commit => '99c75c3f1f54b43492048d6a1e087fcc48f1be6c'
 mod 'pcfens/rhn_register', '1.5.0'
-mod 'puppet/archive', '1.3.0'
 mod 'puppet/yum', '2.1.0'
 mod 'puppet-collectd', '5.3.0'
-mod 'shinesolutions/aem_curator', '3.13.2'
+mod 'shinesolutions/aem_curator', :git => 'https://github.com/hoomaan-kh/puppet-aem-curator.git', :branch => 'jdk11-and-ssl-include-both-java'
```
Comment (mbloch1986) [inline Puppetfile:9]: development left overs ?

File: Puppetfile [hunk 2]
```diff
@@ -4,27 +4,25 @@ mod 'bashtoni/timezone', '1.0.0'
...
-
 # puppet-aem-curator dependencies, TODO: resolve sub-dependencies without declaring deps in packer-aem
-mod 'aco/oracle_java', '2.9.4'
 mod 'bstopp/aem', '3.0.0'
 mod 'puppet-healthcheck', '0.4.0'
 mod 'puppetlabs/apache', :git => 'https://github.com/shinesolutions/puppetlabs-apache', :commit => 'd4f89389faae77e8ef5618069085e429ce50e427'
 mod 'puppetlabs/java_ks', '1.6.0'
-mod 'shinesolutions/aem_resources', '7.0.1'
-
+mod 'shinesolutions/aem_resources', :git => 'https://github.com/hoomaan-kh/puppet-aem-resources.git', :branch => 'jdk11-enable-granite-ssl'
```
Comment (mbloch1986) [inline Puppetfile:23]: development left overs ?

File: templates/ansible/hieradata.j2 [hunk 3]
```diff
@@ -33,15 +33,12 @@ amazon_ssm_agent::proxy_url: {{ http_proxy }}
...
 
 aem_curator::install_author::aem_artifacts_base: {{ aem.artifacts_base }}
 aem_curator::install_author::aem_jvm_mem_opts: "{{ aem.author.jvm_mem_opts }}"
 aem_curator::install_author::aem_jvm_opts:
 - '-XX:+PrintGCDetails'
-- '-XX:+PrintGCTimeStamps'
-- '-XX:+PrintGCDateStamps'
-- '-XX:+PrintTenuringDistribution'
-- '-XX:+PrintGCApplicationStoppedTime'
 - '-XX:+HeapDumpOnOutOfMemoryError'
```
Comment (mbloch1986) [inline templates/ansible/hieradata.j2:42]: this is only required for jdk11 right ? I guess best is to make those parameters configurable, so the user can adjust them to their own need. Though these parameters represent the default values.

File: templates/ansible/hieradata.j2 [hunk 4]
```diff
@@ -53,14 +50,11 @@ aem_curator::install_author::data_volume_device: "{{ aws[os_type].devicemap_data
...
 aem_curator::install_publish::aem_keystore_password: "%{lookup('{{ aem.keystore_password_parameter }}')}"
+aem_curator::install_publish::aem_ssl_method: {{ aem.ssl_method }}
 aem_curator::install_publish::aem_artifacts_base: {{ aem.artifacts_base }}
 aem_curator::install_publish::aem_jvm_mem_opts: "{{ aem.publish.jvm_mem_opts }}"
 aem_curator::install_publish::aem_jvm_opts:
 - '-XX:+PrintGCDetails'
-- '-XX:+PrintGCTimeStamps'
-- '-XX:+PrintGCDateStamps'
-- '-XX:+PrintTenuringDistribution'
-- '-XX:+PrintGCApplicationStoppedTime'
```
Comment (mbloch1986) [inline templates/ansible/hieradata.j2:63]: same as above

File: test/inspec/author_spec.rb [hunk 5]
```diff
@@ -48,13 +52,13 @@
...
-  it { should be_owned_by 'aem-author' }
-  it { should be_grouped_into 'aem-author' }
-end
+# describe file("#{aem_base}/aem/author/crx-quickstart/ssl/aem.ks") do
+#   it { should be_file }
+#   it { should exist }
+#   its('mode') { should cmp '00640' }
+#   it { should be_owned_by 'aem-author' }
+#   it { should be_grouped_into 'aem-author' }
+# end
```
Comment (mbloch1986) [inline test/inspec/author_spec.rb:61]: This test is probably still required if jetty is used as ssl method ?

File: templates/ansible/hieradata.j2 [hunk 6]
```diff
@@ -75,10 +69,13 @@ aem_curator::install_dispatcher::apache_module_tarball: dispatcher-apache2.4-lin
 aem_curator::install_dispatcher::apache_module_filename: dispatcher-apache2.4-{{ aem.dispatcher.version }}.so
 aem_curator::install_dispatcher::data_volume_device: "{{ aws[os_type].devicemap_data_b }}"
 
-aem_curator::install_java::jdk_base_url: "{{ aem.jdk.base_url }}"
-aem_curator::install_java::jdk_filename: "{{ aem.jdk.filename }}"
-aem_curator::install_java::jdk_version: "{{ aem.jdk.version }}"
-aem_curator::install_java::jdk_version_update: "{{ aem.jdk.version_update }}"
+aem_curator::install_java::jdk_base_url: "{{ java.base_url }}"
+aem_curator::install_java::jdk_filename: "{{ java.filename }}"
+aem_curator::install_java::jdk_version: "{{ java.version }}"
```
Comment (mbloch1986) [inline templates/ansible/hieradata.j2:74]: Please update the packer-aem default configuration and add these config parameters. If you add them please refer them to the existing parameters e,g, `{{ aem.jdk.base_url }}` to keep the backward compatibility Thanks

File: test/inspec/author_spec.rb (hunk 5 above)
Comment (hoomaan-kh) [reply test/inspec/author_spec.rb:61]: Yes, it is required

File: templates/ansible/hieradata.j2 (hunk 3 above)
Comment (hoomaan-kh) [reply templates/ansible/hieradata.j2:42]: It is needed for JDK11, I will make this change.

File: templates/ansible/hieradata.j2 (hunk 4 above)
Comment (hoomaan-kh) [reply templates/ansible/hieradata.j2:63]: It is only needed for JDK11, I will make this change.

File: Puppetfile (hunk 1 above)
Comment (hoomaan-kh) [reply Puppetfile:9]: Fixed

File: Puppetfile (hunk 2 above)
Comment (hoomaan-kh) [reply Puppetfile:23]: Fixed

File: templates/ansible/hieradata.j2 (hunk 3 above)
Comment (hoomaan-kh) [reply templates/ansible/hieradata.j2:42]: Done

File: test/inspec/author_spec.rb (hunk 5 above)
Comment (hoomaan-kh) [reply test/inspec/author_spec.rb:61]: Done

File: templates/ansible/hieradata.j2 (hunk 6 above)
Comment (hoomaan-kh) [reply templates/ansible/hieradata.j2:74]: Done

**Comment by mbloch1986** [comment]
I'll close this PR, as we seperated it into 2 different PRs one for adding the new SSL method https://github.com/shinesolutions/packer-aem/pull/223 and one for adding the JDK11 support https://github.com/shinesolutions/packer-aem/pull/222

**Pull request closed without merging**